packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -112,72 +94,60 @@ class PaymentRequestConfirmation extends React.Component<Props> {
 
     const paymentInfo = {
       amount: amount.toString(),
-      comment: this.state.comment || undefined,
+      comment: comment || undefined,
       createdAt: firebase.database.ServerValue.TIMESTAMP,
-      requesterAddress: address,
-      requesterE164Number: this.props.e164PhoneNumber ?? undefined,
+      requesterAddress: account,
+      requesterE164Number: confirmationInput.recipient.e164PhoneNumber ?? undefined,
       requesteeAddress: requesteeAddress.toLowerCase(),
       status: PaymentRequestStatus.REQUESTED,
       notified: false,
     }
 
     ValoraAnalytics.track(RequestEvents.request_confirm_request, { requesteeAddress })
-    this.props.writePaymentRequest(paymentInfo)
+    dispatch(writePaymentRequest(paymentInfo))
     Logger.showMessage(t('requestSent'))
   }
 
-  renderFooter = () => {
-    const amount = {
-      value: this.props.confirmationInput.amount,
-      currencyCode: this.props.confirmationInput.currency,
-    }
-
+  const renderFooter = () => {
```

Comment:
Small nit: I saw you moved the `amount` const to the top of the file. Can we move it back here if it's not used elsewhere? I find it helps readability to have the props used close to where they are defined whenever possible.